ptgnn/neuralmodels/gnn/graphneuralnetwork.py

```diff
@@ -88,12 +95,29 @@ def gnn(
         :param adjacency_lists: a list of [num_edges_per_type, 2] adjacency lists per edge type.
                 The order is fixed across runs. Backwards edges and self-edges are included if
                 the appropriate hyperparameter is set.
+        :param edge_feature_embeddings: a list of the edge features per edge-type.
         :param node_to_graph_idx: A mapping that tells us which graph the node belongs to
         :param reference_node_ids: A dictionary indicating the reference node index
         :param reference_node_graph_idx: A dictionary indicating the graph index for reference node
         :param return_all_states: Whether to return all states
         :return: a [num_nodes, output_hidden_dimension] matrix of the output representations
         """
+        if self.__edge_dropout_rate > 0 and self.training:
+            dropped_adj_list, dropped_edge_features = [], []
+            for (edge_sources_idxs, edge_target_idxs), edge_features in zip(
+                adjacency_lists, edge_feature_embeddings
+            ):
+                mask = (
+                    torch.rand_like(edge_sources_idxs, dtype=torch.float32)
+                    > self.__edge_dropout_rate
+                )
+                dropped_adj_list.append(
+                    (edge_sources_idxs.masked_select(mask), edge_target_idxs.masked_select(mask))
+                )
+                dropped_edge_features.append(edge_features[mask])
```

Comment:
The standard dropout implementation usually also does rescaling (see e.g. https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html). Do that here as well?